students/CCSt130/lesson03/mail_room_pt1.py

```diff
@@ -0,0 +1,500 @@
+        if not donor[:][3] in donor_names:
+            #donor[:][3] = temp_donor 
+            donor_names.append(donor[:][3])
+    
+    donor_name_list = []
+
+    temp_donor_list = []
+    
+    donation_count = []
+    
+    # Put each instance of a donor name into a list
+    # This will allow collection of each donation amount
+    for donor in donations:
+        temp_donor_list.append(donor[:][3])
+        
+    # Put Counter output into a name
+    # To determine count of each donor's donations
+    donor_name_list = Counter(temp_donor_list)
+    
+    # Get into a list of tuples
+    donation_count = donor_name_list.most_common()
+    
+    # Sort list of donor tuples
+    donation_count.sort()
+
+    print()
+    header2 = ['Donor', 'CT', 'TTL Given', 'AVG Gift']
+    print()
+    # Print header
+    print('{:<34} {:<3} {:<16} {}'.format(*header2))
+    print(70 * '-')
+
+    for ea_name in donor_names:
+
+        temp_list = []
+
+        for donor in donations:
+            if donor[:][3] == ea_name:
+                # Append donation amounts found for individual donor to a list
+                temp_list.append(donor[:][4])
+                # Count or sum of tl individual donations for each donor
+                donation_ctr = (len(temp_list))
+
+                donation_ttl = (sum(temp_list))
+                
+                # Find the average of the donations
+                donation_avg = mean(temp_list)
+
+                search = ea_name
+                
+                for sublist in donation_count:
+                    if sublist[0] == search:
+                        if((sublist[1]) == (donation_ctr)):
+                            # Replace below with formatted string
+                            """
+                            print("Result: ", end = "")
+                            print(ea_name, end = " ")
+                            print(donation_count, end = " ")
+                            print(donation_avg)
+                            """
+                            # Print formatted string of donations
+                            print('{:<34} {:>2}  ${:>14,.2f}  ${:>13,.2f}'.format(ea_name, donation_ctr, donation_ttl, donation_avg))
+    print()
+
+def list_donor_names(all_donations):
+    """ Collects a list of unique individual donor names. """
+    # List holds donations from donor selected
+    donor_names = []
+    # List holds enumerated list
+    donor_list = []
+    
+    for donor in all_donations:
+        if not donor[:][3] in donor_names:
+            # Add donor name only to new list
+            donor_names.append(donor[:][3])
+
+    # Convert to numbered list
+    num_list = enumerate(donor_names,1)
+    
+    print()
+    print("Our Donors:")
+    print()
+    
+    # Convert enumerated list to tuple and print
+    for names in num_list:
+        donor_list.append(names)
+        print(f' {names[0]}. {names[1]}')
+           
+    print()
+    # Return tuple
+    return(donor_list)
```

Comment:
none of your functions that are called from my entry point need `return` statements.